Yoast/wordpress-seo
# No longer touch the network admin if the plugin is not network-active

## Summary

This PR can be summarized in the following changelog entry:

* Fixes a bug where the Yoast SEO network admin menu and admin bar menu would appear when the plugin was only active for the main site and not for the entire network.

## Relevant technical choices:

* When a plugin is active for the main site, it is essentially also active in the context of the network admin, even when it is not network active. That is because the network admin loads in scope of the main site.
* While this is mostly a problem in WordPress, it's unlikely that it can and will be solved in a proper way soon, so we should manually ensure that Yoast SEO does not touch the network admin when the plugin is not network-active.
* A new `WPSEO_Utils::is_plugin_network_active()` method has been introduced. It returns true when the plugin is either among the network active plugins or when it is an MU-plugin (which means it is active for the entire network too, more precisely the entire setup).
* This method is used in the necessary places to prevent the following three components from initializing:
    * the network admin menu
    * the network admin bar menu
    * the network admin functionality

## Test instructions

This PR can be tested by following these steps:

* Setup a multisite.
* Activate the plugin on the main site (if you have only one site, that is the main site).
* Go to the network admin.
* Ensure that the Yoast SEO admin menu and admin bar menu do not show.
* Deactivate the plugin again on the main site.
* Go to the network admin and activate the plugin from there (thus for the entire network).
* Ensure that the Yoast SEO admin menu and admin bar menu show.

## Quality assurance


Fixes #2443
